modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java

```diff
@@ -283,16 +285,19 @@ private List<BundleDependency> resolveSystemScopeDependencies(AetherMavenClient
   }
 
   private List<BundleDependency> getUniqueDependencies(List<BundleDependency> dependencies) {
-    Map<String, BundleDependency> uniqueDependencies = new HashMap<>();
+    Set<String> uniqueDependenciesIds = new HashSet<>();
 
-    for (BundleDependency dependency : dependencies) {
+    // Filtering is done this way to preserve the order
+    return dependencies.stream().filter(dependency -> {
       org.mule.maven.client.api.model.BundleDescriptor descriptor = dependency.getDescriptor();
       String pluginKey =
-          descriptor.getGroupId() + ":" + descriptor.getArtifactId() + ":" + descriptor.getClassifier().orElse("");
-      uniqueDependencies.putIfAbsent(pluginKey, dependency);
-    }
-
-    return new ArrayList<>(uniqueDependencies.values());
+          descriptor.getGroupId() + ":" + descriptor.getArtifactId() + ":" + descriptor.getVersion()
+              + descriptor.getClassifier().map(classifier -> ":" + classifier).orElse("");
+      boolean isApi = descriptor.getClassifier().map(API_CLASSIFIERS::contains).orElse(false);
```

Comment:
deprecate ArtifactConstants#API_CLASSIFIERS since it can easily become outdated. Provide a utility method that gets those classifiers form a system property